metadata-ingestion/tests/integration/grafana/test_grafana.py

```diff
@@ -273,6 +282,51 @@ def verify_grafana_fully_ready(
     logging.warning(f"Grafana may not be fully ready after {timeout}s timeout")
 
 
+def verify_grafana_entities_provisioned(timeout: int = 180) -> None:
+    """Wait for Grafana entities to be provisioned before running ingestion tests
+
+    This function can be extended in the future to validate other entity types
+    like data sources, folders, etc.
+    """
+    base_url = "http://localhost:3000"
+
+    session = requests.Session()
+    retry_strategy = Retry(
+        total=3, backoff_factor=1, status_forcelist=[500, 502, 503, 504]
+    )
+    adapter = HTTPAdapter(max_retries=retry_strategy)
+    session.mount("http://", adapter)
```

Comment:
can we avoid having the base url and session setup stuff repeated? looks like this is copy-pasted from the above method